client/src/components/useraccess/Forms/PasswordResetModal.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from '../../userprofile/adminSubComponents/loader';
+import * as userActions from '../../../Actions/userAccessAction';
+
+/**
+ * @class PasswordResetModal
+ * @classdesc returns Password reset modal
+ */
+class PasswordResetModal extends React.Component {
+  constructor(prop) {
+    super();
+    this.state = {
+      email: '',
+      loader:false,
+      disableBtn: false
+    }
+
+    this.handleInput = this.handleInput.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.cancelPassReset = this.cancelPassReset.bind(this);
+    const message = document.getElementById('message');
+  }
+
+  /**
+   * 
+   * @param { object } event 
+   * @memberof PasswordResetModal
+   * @returns { object } a new state with use email
+   */
+  handleInput(event) {
+    this.setState({email: event.target.value})
+  }
+
+  /**
+   * 
+   * 
+   * @param { object } event 
+   * @memberof PasswordResetModal
+   * @returns { object } respose from api call 
+   */
+  handleSubmit(event) {
+    // validate input
+    this.setState({ error: true, disableBtn: true});
+    if(this.state.email === "") {
+    message.innerHTML = `Your email is needed to reset your password`;
+    message.className="red-text center-align";  
+    } else {
+      
+      //Make API call
+      this.props.sendEmail({email: this.state.email})
+      .then(() => {
+        this.setState({ error: false, disableBtn: true});
+        message.innerHTML = `A password reset link has been sent to ${this.state.email}`;
+        message.className="green-text center-align";
+      })
+      .catch((error) => {
+        if (error.response.status === 501) {
+          message.innerHTML ="Ooops! Something went wrong, the server could not process your request at this time. Please try again.";
+          message.className="red-text center-align";
+          this.setState({ error: false, disableBtn: false});
+        } else {
+          message.innerHTML = error.response.data.message;
+          message.className="red-text center-align";
+          this.setState({ error: false, disableBtn: false});
+        }
+      })
+    }
+    // prevent button default action
+    event.preventDefault();
+  }
+
+  cancelPassReset() {
+    this.setState({email: '', loader: false, disableBtn: false
+  });
+    message.innerHTML = 'A password reset link will be sent to this email';
+    message.className="black-text center-align"
+  }
+
+  render() {
+    return(
+    <div className="modal passwordModal" id="passwordResetModal">
+      <div className="container">
+        <div className="col s12 m6 l4 offset-l8">
+          <div className="modal-content">
+            <h5 className="center-align">Password reset</h5>
+              <p className="center-align" id="message">
+              A password reset link will be sent to this email</p>
+              {
+                this.state.error 
```

Comment:
The variable error is kinda confusing, I think loading or processing would have been better